core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -693,6 +693,22 @@ public static boolean lt(BigDecimal b0, BigDecimal b1) {
     return b0.compareTo(b1) < 0;
   }
 
+  /** SQL <code>&lt;</code> operator applied to List<Comparable> values. */
+  public static <T extends Comparable> boolean lt(List<T> b0, List<T> b1) {
+    for (int i = 0; i < b0.size(); i++) {
+      // b0 is greater because b0 is longer than b1.
```

Comment:
From your changes, it seems that the array comparison is allowed in the logic layer, we just do not have a right implementation ~

Reply:
Yep!